src/internal.rs

```diff
@@ -728,11 +739,35 @@ pub(crate) struct Rav1dTaskContext {
     pub task_thread: Arc<Rav1dTaskContext_task_thread>,
 }
 
-// TODO(SJC): This is a temporary struct to pass a single pointer that holds
-// both a Rav1dContext and Rav1dTaskContext to the start routine in
-// pthread_create. We need to pass the Rav1dTaskContext into the thread by value
-// and remove it from the context structure.
-pub(crate) struct Rav1dTaskContext_borrow<'c> {
-    pub c: &'c Rav1dContext,
-    pub tc: &'c mut Rav1dTaskContext,
+impl Rav1dTaskContext {
+    pub(crate) unsafe fn new(
+        f: *const Rav1dFrameContext,
+        task_thread: Arc<Rav1dTaskContext_task_thread>,
+    ) -> Self {
+        let mut cf = mem::zeroed();
+        *BitDepth16::select_mut(&mut cf) = Align64([0; 32 * 32]);
+        Self {
+            f,
+            ts: ptr::null_mut(),
+            bx: 0,
+            by: 0,
+            l: mem::zeroed(),
+            a: ptr::null_mut(),
+            rt: mem::zeroed(),
+            cf,
+            al_pal: Default::default(),
+            pal_sz_uv: Default::default(),
+            txtp_map: [0u8; 1024],
+            scratch: mem::zeroed(),
+            warpmv: mem::zeroed(),
```

Comment:
We might want to `#[derive(Zeroable)]` (using `bytemuck` or something similar) these `union`s to more safely initialize them.  This is clearly sound for now, but if in the future some sub-field becomes an `enum`, it becomes UB, for example.  Not important to do right here, but we should in the future.

Reply:
Yeah, I think I'd prefer we impl Default or provide initializers rather than adding bytemuck for this.

Reply:
I meant `impl Default` through `impl Zeroable` using `bytemuck`'s auto derive, since otherwise it's easy to get wrong and not initialize the largest `union` variant or something.

Reply:
Understood. Yeah, I still don't want to add a dependency for that right now.